tuta-sdk/rust/sdk/src/entity_client.rs

```diff
@@ -23,6 +23,12 @@ pub struct EntityClient {
 	type_model_provider: Arc<TypeModelProvider>,
 }
 
+impl EntityClient {
+	pub fn get_headers_provider(&self) -> &Arc<HeadersProvider> {
```

Comment:
why is this necessary? Why is it reference to `Arc`?

Reply:
Arc

Reply:
should be fixed, removed the impl block